opencog/atomspace/atom_types.script

```diff
@@ -76,6 +76,9 @@ EXISTS_LINK <- BIND_LINK
 AVERAGE_LINK <- FORALL_LINK
 SATISFYING_SET_LINK <- BIND_LINK
 
+// Function constructor (predicate and schema)
+LAMBDA_LINK <- BIND_LINK
```

Comment:
I don't really understand the meaning of BindLink to be honest. The way I understood it is, a link of the type

BindLink
    Variables
    Expression

No more than that. But a lambda is a bit more than that, it abstract away an expression to turn it into a function (that can then be used for application). And a quantifier is not a lambda, but it also has the same binding pattern in common. So that is why I thought it makes sense to have both lambda and quantifiers inherits bindLink.

As for SatisfyingSetLink I don't think it should do any binding at all.

Rather than renaming BindLink by GroundingLink, perhaps you want to add a new link GroundingLink that inherits BindLink, and that would be used by the pattern matcher, that makes sense to me.